### PR TITLE
Check for change in voltage and encoder positions

### PR DESCRIPTION
**Pull request by Roelemans**
## Description

I made the naïve assumption that imc voltage would always change. I forgot that imc voltage, is contained in 16 bits and therefore highly discrete. So instead, we should check if any of imc voltage, or the encoders have updated.

## Changes

* Check whether either voltage or one of the encoders has updated instead of voltage alone.

### CODE REVIEW

File: march_hardware/include/march_hardware/Joint.h [hunk 1]
```diff
@@ -122,6 +122,8 @@ class Joint
   const int net_number_;
   bool allow_actuation_ = false;
   float previous_imc_volt_ = 0.0;
+  double previous_absolute_position = 0.0;
+  double previous_incremental_position = 0.0;
```
Comment (Olavhaasie) [inline march_hardware/include/march_hardware/Joint.h]: :see_no_evil: 
```suggestion
  double previous_absolute_position_ = 0.0;
  double previous_incremental_position_ = 0.0;
```

**Review by Olavhaasie — CHANGES_REQUESTED** [review]

File: march_hardware/include/march_hardware/Joint.h (hunk 1 above)
Comment (Roelemans) [reply march_hardware/include/march_hardware/Joint.h]: Oopsie

**Review by Olavhaasie — DISMISSED** [review]

**Review by RutgerVanBeek — CHANGES_REQUESTED** [review]
You still have a chance (although small) of false positives with this technique. It seems that the current  the most unique values. Therefore I think you should at least include the current in the test. Is there by any chance a PDO mappable object that should be unique be definition, like time, that we could use for this?

**Review by Joris-JW — DISMISSED** [review]
I think the chance of false positives is significantly small.

**Comment by Roelemans** [comment]
The test now also check for current. As Rutger mentioned, it would be ideal if there is a variable in the PDO that always changes, like time. We coincidentally also want to add time to the PDO for encoder data filters, so we will look at that in the future.

**Review by RutgerVanBeek — DISMISSED** [review]

**Review by RutgerVanBeek — APPROVED** [review]

**Review by Joris-JW — APPROVED** [review]

**Pull request merged**